bundles/org.openhab.binding.vdr/src/main/java/org/openhab/binding/vdr/internal/VDRHandler.java

```diff
@@ -0,0 +1,357 @@
+                updateStatus(ThingStatus.OFFLINE);
+            }
+
+            scheduleRefreshThread();
+        });
+    }
+
+    /**
+     * Update Thing's properties (e. g. VDR Version)
+     *
+     * @param client the {@link SVDRPClient} to be used for properties update
+     */
+    public void updateProperties(SVDRPClient client) {
+        Map<String, String> properties = editProperties();
+        // set vdr version to properties of thing
+        String version = client.getSVDRPVersion();
+        properties.put("version", version.toString());
+
+        // persist changes only if there are any changes.
+        if (!editProperties().equals(properties)) {
+            updateProperties(properties);
+        }
+    }
+
+    /**
+     * Handling Commands for {@link VDRHandler}
+     *
+     * @param channelUID channel command was executed for
+     * @param command command to execute
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        SVDRPClient con = new SVDRPClientImpl(config.getHost(), config.getPort());
+        try {
+            con.openConnection();
+
+            if (command instanceof RefreshType) {
+                this.onVDRRefresh();
+            } else {
+                State result = UnDefType.NULL;
+                String cmd = command.toString();
+                switch (channelUID.getId()) {
+                    case "message":
+                        con.sendSVDRPMessage(cmd);
+                        break;
+                    case "power":
+                        con.sendSVDRPKey(cmd);
+                        break;
+                    case "channel":
+                        SVDRPChannel channel = con.setSVDRPChannel(Integer.parseInt(cmd));
+                        result = new DecimalType(channel.getNumber());
+                        updateState(channelUID, result);
+                        break;
+                    case "volume":
+                        SVDRPVolume volume = con.setSVDRPVolume(Integer.parseInt(cmd));
+                        result = new PercentType(volume.getVolume());
+                        updateState(channelUID, result);
+                        break;
+                    case "keyCode":
+                        con.sendSVDRPKey(cmd.trim());
+                        break;
+                }
+            }
+        } catch (SVDRPParseResponseException e) {
+            logger.trace("VDR handleCommand for Thing {}, ChannelUID {}, Parse Response failed. Message: {}",
+                    this.getThing().getUID(), channelUID, e.getMessage());
+        } catch (SVDRPConnectionException e) {
+            logger.debug("VDR handleCommand for Thing {}, ChannelUID {}, Connection failed. Message: {}",
+                    this.getThing().getUID(), channelUID, e.getMessage());
+        } catch (
+
+        Exception ex) {
+            logger.error("There is something wrong with your thing '{}', please check or recreate it: {}",
```

Comment:
Bindings should only log to error if something severe happened, like the detection of a bug in your code. This could be debug. See this link for a description of the log levels: https://www.openhab.org/docs/developer/guidelines.html#f-logging